wordpress-mobile/WordPress-Android
# Remove unnecessary errors from missing jetpack version

I've noticed that we were having an exception in the logs that was saying that Jetpack version is illegal. This is happening on my site when the jetpack version is "false". Checking the code that produces this exception there is a very weird behaviour that catches the exception and returns "true" (meaning that the site has the correct jetpack version) when the version parsing fails. I've changed this behaviour because I'm pretty sure it's wrong.
I've added unit tests to test the changed behaviour.

To test:
* Test the app with a self-hosted site (jurassic.ninja with and without Jetpack)
* Check that the plugins item is visible only when it should be
* Check that there is no error in the Logcat

PR submission checklist:
